pkg/debuginfo/metadata.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2022 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debuginfo
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"path"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/thanos-io/objstore"
+)
+
+var (
+	logger                                     log.Logger
+	ErrDebugInfoMetadataShouldExist            = errors.New("debug info metadata should exist")
+	ErrDebugInfoMetadataExpectedStateUploading = errors.New("debug info metadata state should be uploading")
+)
+
+type metadataState int64
+
+const (
+	// There was an unexpected error. The error will be filled in in the return
+	// value
+	metadataStateError metadataState = iota
+	// There's no debug info metadata. This could mean that an older Parca version
+	// uploaded the debug info files, but there's not record of the metadata, yet.
+	metadataStateEmpty
+	// The debug info file is being uploaded.
+	metadataStateUploading
+	// The debug info file is fully uploaded.
+	metadataStateUploaded
+)
+
+func setMetadataLogger(l log.Logger) {
```

Comment:
Let's create a `MetaDataManager` to manage the state rather than using package-level variables. Which can store the `logger` and the `bucket`.
We can initialize an instance per debuginfo.Store.